### PR TITLE
Add snap support for docker interface for app dev builds

### PR DESCRIPTION
**Pull request by nicktate**
#### What does this pull request accomplish

Adds `app-dev-build` plug w/ the `docker` interface

> This will require the user to have docker installed via snap and execute `snap connect doctl:app-dev-build docker:docker-daemon` to grant permissions

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -35,7 +35,8 @@ parts:
     # Override to prevent 'invalid cross-device link' calling doctl
     - XDG_CONFIG_HOME: "$SNAPCRAFT_PART_INSTALL/tmp/.config"
     override-pull: |
-      git clone https://github.com/digitalocean/doctl.git .
+      git clone --branch ntate/appdev/snapcraft https://github.com/nicktate/doctl.git .
```
Comment (andrewsomething) [inline snap/snapcraft.yaml:38]: Just flagging that this will need to be removed before merging to main.

**Review by andrewsomething — APPROVED** [review]
:+1:  LGTM!

**Review by kamaln7 — APPROVED** [review]

**Comment by andrewsomething** [comment]
Also wanted to point out that we do have a few places where we do some custom error messages to point snap users in the right direction. E.g.

https://github.com/digitalocean/doctl/blob/51c4f6b63a5f6dd7c29529c1df4e06350d592779/commands/registry.go#L425-L429

Might want to consider doing that for app dev builds as well.

**Comment by nicktate** [comment]
> Also wanted to point out that we do have a few places where we do some custom error messages to point snap users in the right direction. E.g.
> 
> https://github.com/digitalocean/doctl/blob/51c4f6b63a5f6dd7c29529c1df4e06350d592779/commands/registry.go#L425-L429
> 
> Might want to consider doing that for app dev builds as well.

💯 @kamaln7 and I were just talking about this as well. I will add it in, thanks for the reference

**Pull request merged**